metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorFactory.java

```diff
@@ -95,18 +93,37 @@ public HiveConnectorFactory(
             throw new IllegalArgumentException(
                 String.format("Failed creating the hive metastore client for catalog: %s", catalogName), e);
         }
-        final Module hiveModule = new HiveConnectorModule(config,
-            catalogName, configuration, infoConverter, client, registry);
-        this.injector = Guice.createInjector(hiveModule);
+        final ApplicationContext ctx = new AnnotationConfigApplicationContext(HiveConnectorConfig.class);
```

Comment:
Here is where I think you should add the connector name and other things as beans and then call `refresh` on the context.